### PR TITLE
[MPMD-349] Upgrade Maven Reporting API to 3.1.1/Maven Reporting Impl to 3.2.0

### PR DESCRIPTION
**Pull request by michael-o**
Following this checklist to help us incorporate your
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MPMD) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue.  Your pull request should address just this issue, without
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MPMD-XXX] - Subject of the JIRA Ticket`,
       where you replace `MPMD-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [ ] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by michael-o** [comment]
Guys, JXR 3.3.0 is out, please review.

File: src/site/apt/usage.apt.vm [hunk 1]
```diff
@@ -74,7 +74,7 @@ Configuration
   The reports will link directly to the cross-referenced source if you enable this with the <<<linkXRef>>> parameter.
   See the {{{http://maven.apache.org/plugins/maven-jxr-plugin/}JXR plugin}} for more details.
 
-  If your source uses a non-default encoding, you can use the <<<sourceEncoding>>> parameter to tell Maven which
+  If your source uses a non-default encoding, you can use the <<<inputEncoding>>> parameter to tell Maven which
   encoding to use when reading the java source.  Note also the ability to exclude source which you want
   to ignore.
```
Comment (slawekjaranowski) [inline src/site/apt/usage.apt.vm:79]: why such change?
Is it not braking?

Comment (michael-o) [reply src/site/apt/usage.apt.vm:79]: This basically as in https://github.com/apache/maven-checkstyle-plugin/pull/87#discussion_r950316588. We would end up in two params for encoding.

**Review by slawekjaranowski — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #319

**Pull request merged**